pkg/obfuscate/sql.go

```diff
@@ -287,13 +287,25 @@ func (f *groupingFilter) Reset() {
 	f.groupMulti = 0
 }
 
+func isSQLLexer(obfuscationMode ObfuscationMode) bool {
+	return obfuscationMode != ""
+}
+
 // ObfuscateSQLString quantizes and obfuscates the given input SQL query string. Quantization removes
 // some elements such as comments and aliases and obfuscation attempts to hide sensitive information
 // in strings and numbers by redacting them.
 func (o *Obfuscator) ObfuscateSQLString(in string) (*ObfuscatedQuery, error) {
 	return o.ObfuscateSQLStringWithOptions(in, &o.opts.SQL)
 }
 
+// ObfuscateSQLStringForDBMS quantizes and obfuscates the given input SQL query string for a specific DBMS.
+func (o *Obfuscator) ObfuscateSQLStringForDBMS(in string, dbms string) (*ObfuscatedQuery, error) {
+	if isSQLLexer(o.opts.SQL.ObfuscationMode) {
+		o.opts.SQL.DBMS = dbms
```

Comment:
Aren't there some variants in the DBMS string that we get from the span meta information? i.e. `postgres` vs. `postgresql`. It doesn't look like `go-sqllexer` accounts for these, do we need some mapping logic before setting that here?

Reply:
In the Lexer, we map .Net's `sqlserver` and Java's `sql-server` to mssql. I'll add `postgres/postgresql` mapping as well, but that isn't a blocker for this PR.

Reply:
[postgres alias in sqllexer](https://github.com/DataDog/go-sqllexer/pull/49)